### DISCUSSION TITLE
Sending function definitions from Header files to Copilot for C/C++

### DISCUSSION THREAD

**Discussion by pratham1002** [community/community#14924] [Copilot Conversations]
Copilot performance may be improved if it receives the function definitions in the Custom Headers (user-defined .h files) for C and C++. This would give the model information about the function signature for functions, not in the standard library. For medium-sized projects, just concatenating function definitions (which can be picked up from the C/C++ intellisense supported by VS code) should be tested and might be a fair tradeoff against some loss of local context.

A simple example case: defining a `linkedlist` in C.

```C
// linkedlist.h

struct node {
    int val;
    struct node next;
}

struct node* newNode(int value);
void append(int value, struct node* head);
```
```C
// linkedlist.c

void append(int value, struct node* head)
{
   // traverse till the of the list and append a new node
   // Now expecting completion from Copilot to make correct use of the newNode function for which it needs the signature.
}
```

**Comment by Kek5chen** [comment]
We do not really know what it is sending to the server and if it differs from IDE to IDE.

I am guessing you are using VS Code, I use JetBrains products and dont have any problem with this, but that could just be my use case.

**Reply by pratham1002** [reply]
I am working on a toy compiler as an academic project, as the code is split into a lot of modules that we import across files, copilot struggles to use the functions correctly and nearly always messes up the sequence of function arguments.
